File: Leetcode/1695. Maximum Erasure Value.py
class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        start, end, currSum, maxSum, N, Set = 0, 0, 0, 0, len(nums), set()
        while end < N and start <= end:
            if nums[end] in Set:
                while nums[start] != nums[end]:
                    currSum -= nums[start]
                    Set.remove(nums[start])
                    start += 1
                start += 1
            else:
                currSum += nums[end]
                maxSum = max(maxSum, currSum)
                Set.add(nums[end])
            end += 1
        return maxSum

File: Leetcode/1695. Maximum Erasure Value.cpp
class Solution {
public:
    int maximumUniqueSubarray(vector<int>& nums) {
        int i = 0, j = 0, sum = 0, maxSum = 0, N = nums.size();
        set <int> Set;
        while (j < N && i <= j) {
            if (Set.find(nums[j]) == Set.end()) {
                sum += nums[j];
                maxSum = max(maxSum, sum);
                Set.insert(nums[j++]);
            } else {
                while (nums[i] != nums[j]) {
                    sum -= nums[i];
                    Set.erase(Set.find(nums[i++]));
                }
                i++;
                j++;
            }
        }
        return maxSum;
    }
};